### PR TITLE
change person hash data detection - work around

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

This PR includes the following proposed change(s):

- change person hash data detection - - work around

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested locally

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs [hunk 1]
```diff
@@ -659,5 +663,30 @@ private async Task<EmploymentEntity> LinkEmploymentRef(EmploymentEntity employme
...
+                return existedPerson;
+            }
+            catch (WebRequestException e)
+            {
+                if(e.Code== HttpStatusCode.NotFound)
+                {
+                    return null;
+                }
+                throw;
+            }catch(Exception)
```
Comment (pbolduc) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs:686]: Dont need this extra catch

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs [hunk 2]
```diff
@@ -659,5 +663,30 @@ private async Task<EmploymentEntity> LinkEmploymentRef(EmploymentEntity employme
...
+            string hashData = newPerson.DuplicateDetectHash;
+            try
+            {
+                var existedPerson = await this._oDataClient.For<SSG_Person>()
+                    .Filter(x => x.DuplicateDetectHash == hashData)
+                    .FindEntryAsync(CancellationToken.None);
+                    
+                return existedPerson;
+            }
+            catch (WebRequestException e)
```
Comment (pbolduc) [inline app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs:679]: Can we use exception filter here  `catch (WebRequestException e) when (e.Code == HttpStatusCode.NotFound)`

**Review by pbolduc — CHANGES_REQUESTED** [review]

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs:686]: ok. changed

File: app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs:679]: agree. code changed.

**Review by pbolduc — APPROVED** [review]
LGTM

**Review by ChrisHoban — APPROVED** [review]
Looks good

**Pull request merged**